extra_data/write_cxi.py

```diff
@@ -92,89 +131,96 @@ def _map_chunk(self, chunk, chunk_data, target, tgt_ax1, have_data=None):
             # Look up where the start of this chunk fits in the target
             tgt_start = int(self.train_id_to_ix[chunk_tids[0]])
 
-            target_tids = self.train_ids_perframe[tgt_start : tgt_start+len(chunk_tids)]
-            assert target_tids.shape == chunk_tids.shape, \
-                f"{target_tids.shape} != {chunk_tids.shape}"
+            # Chunk train IDs organized in the same way as in the target
+            chunk_tids_target = np.repeat(chunk_tids, self.ncells)
+
+            target_tids = self.train_ids_perframe[
+                tgt_start : tgt_start + len(chunk_tids)*self.ncells
+            ]
+            assert target_tids.shape == chunk_tids_target.shape, \
+                f"{target_tids.shape} != {chunk_tids_target.shape}"
             assert target_tids[0] == chunk_tids[0], \
                 f"{target_tids[0]} != {chunk_tids[0]}"
 
             # How much of this chunk can be mapped in one go?
-            mismatches = (chunk_tids != target_tids).nonzero()[0]
+            mismatches = (chunk_tids_target != target_tids).nonzero()[0]
             if mismatches.size > 0:
-                n_match = mismatches[0]
+                n_match = int(mismatches[0] / self.ncells)
```

Comment:
BTW, you can do integer division (aka floor division) with the `//` operator.

Reply:
Thank you, changed in https://github.com/European-XFEL/EXtra-data/commit/6582e9954511eb0bc752851d8f64a31bb7ed1348.